test/model/content_test.dart

```diff
@@ -1,18 +1,18 @@
 import 'package:test/test.dart';
 import 'package:zulip/model/content.dart';
 
+import 'content_matchers.dart';
+
 void main() {
   test('parse a plain-text paragraph', () {
-    // TODO find a way to make these more concise, so we can more easily have many of them
-    final result = parseContent('<p>hello world</p>');
-    expect(result.nodes.length, 1);
-    final node = result.nodes.single;
-    expect(node, const TypeMatcher<ParagraphNode>());
-    expect((node as ParagraphNode).wasImplicit, false);
-    expect(node.nodes.length, 1);
-    final span = node.nodes.single;
-    expect(span, const TypeMatcher<TextNode>());
-    expect((span as TextNode).text, 'hello world');
+    // TODO try to compact this further
+    expect(
+      parseContent('<p>hello world</p>'),
+      ZulipContentMatcher([
+        ParagraphNodeMatcher(
+          wasImplicit: equals(false),
```

Comment:
Possibly this could be made more compact by letting the caller choose between a matcher and a literal value (with the `equals` matcher implied), so you could pass `wasImplicit: false` instead of `wasImplicit: equals(false)`. Then internally `ParagraphNodeMatcher` could use [`wrapMatcher`](https://pub.dev/documentation/matcher/latest/matcher/wrapMatcher.html).